### PR TITLE
Make eext builds reproducible

### PR DESCRIPTION
**Pull request by aajith-arista**
Using the standard methods that make rpmbuild reproducible.

The idea here is to use the SOURCE_DATE_EPOCH environment to set build times, and the _buildhost rpm macro to freeze the build host.

We set the following macros to 1 and enable the behavior described below:
1. source_date_epoch_from_changelog: SOURCE_DATE_EPOCH is set by rpmbuild to match the the latest changelog entry in the spec file. Note that if there's no changelog SOURCE_DATE_EPOCH is not set. It's upto the users to make sure there's a changelog entry.
2. use_source_date_epoch_as_buildtime: This sets the build time as per thr variable.
3. clamp_mtime_to_source_date_epoch: All file mtimes are clamped to this value.

References:
https://rpm-software-management.github.io/rpm/manual/buildprocess.html#Reproducability https://fedoraproject.org/wiki/Changes/ReproducibleBuildsClampMtimes https://github.com/rpm-software-management/mock/issues/692

Fixes: BUG925340

### CODE REVIEW

File: configfiles/mock.cfg.template [hunk 1]
```diff
@@ -8,11 +8,34 @@ config_opts['use_bootstrap_image'] = False
...
+# If true, set the SOURCE_DATE_EPOCH environment variable
+# to the timestamp of the topmost changelog entry
+config_opts['macros']['source_date_epoch_from_changelog'] = '1'
+
+# If true, make sure that buildtime in built rpms
+# is set to the value of SOURCE_DATE_EPOCH.
+# Is ignored when SOURCE_DATE_EPOCH is not set.
+config_opts['macros']['use_source_date_epoch_as_buildtime'] = '1'
+
+
```
Comment (manith-arista) [inline configfiles/mock.cfg.template:28]: Extra new line can be removed

Comment (aajith-arista) [reply configfiles/mock.cfg.template]: Done

**Comment by manith-arista** [comment]
Changes to create reproducible eext builds looks good.

**Comment by aajith-arista** [comment]
> Changes to create reproducible eext builds looks good.

Can you pls approve. Now the branch protection rules mandates approval

**Review by manith-arista — APPROVED** [review]
Changes look good, ship it.

**Pull request merged**